src/pages/vaults/VaultLockedSharesTable.tsx

```diff
@@ -0,0 +1,121 @@
+import { useMemo, useState } from 'react';
+
+import { sum } from 'lodash';
+import styled from 'styled-components';
+
+import { VaultShareUnlock } from '@/constants/abacus';
+import { ButtonShape, ButtonSize } from '@/constants/buttons';
+import { STRING_KEYS } from '@/constants/localization';
+
+import { useApiState } from '@/hooks/useApiState';
+import { useStringGetter } from '@/hooks/useStringGetter';
+import { useLoadedVaultAccount } from '@/hooks/vaultsHooks';
+
+import { tradeViewMixins } from '@/styles/tradeViewMixins';
+
+import { Button } from '@/components/Button';
+import { Output, OutputType } from '@/components/Output';
+import { ColumnDef, Table } from '@/components/Table';
+
+import { runIf } from '@/lib/do';
+
+export const VaultLockedSharesCard = ({ className }: { className?: string }) => {
+  const stringGetter = useStringGetter();
+  const [showShares, setShowShares] = useState(false);
+  const vaultAccount = useLoadedVaultAccount().data;
+  const rawLockedShares = vaultAccount?.vaultShareUnlocks;
+  const lockedShares = useMemo(() => rawLockedShares?.toArray(), [rawLockedShares]);
+  const lockedSharesTotalValue = useMemo(
+    () => sum(lockedShares?.map((s) => s.amountUsdc)),
+    [lockedShares]
+  );
+
+  if (lockedShares == null || lockedShares.length === 0) {
+    return null;
+  }
+  return (
+    <div className={className} tw="rounded-[0.7rem] border border-solid border-color-border">
+      <div tw="flex justify-between px-1 py-0.625">
+        <div tw="flex gap-0.5">
+          <h3 tw="pt-[5px]">Locked Balance</h3>
+          <span tw="text-color-text-0">
+            <Output value={lockedSharesTotalValue} type={OutputType.CompactFiat} />
+          </span>
+        </div>
+        <Button
+          size={ButtonSize.XSmall}
+          shape={ButtonShape.Pill}
+          onClick={() => setShowShares((o) => !o)}
+        >
+          {showShares
+            ? stringGetter({ key: STRING_KEYS.HIDE })
+            : stringGetter({ key: STRING_KEYS.VIEW })}
+        </Button>
+      </div>
+      {showShares && <VaultLockedSharesTable lockedShares={lockedShares} />}
+    </div>
+  );
+};
+
+const VaultLockedSharesTable = ({
+  className,
+  lockedShares,
+}: {
+  className?: string;
+  lockedShares: VaultShareUnlock[];
+}) => {
+  const stringGetter = useStringGetter();
+  const { height } = useApiState();
+
+  const columns = useMemo<ColumnDef<VaultShareUnlock>[]>(
+    () =>
+      [
+        {
+          columnKey: 'time',
+          getCellValue: (row) => row.unlockBlockHeight,
+          label: stringGetter({ key: STRING_KEYS.AVAILABLE }),
+          renderCell: ({ unlockBlockHeight }) => {
+            const estimatedUnlockMs = runIf(
```

Comment:
I like this, no more != null checks, just create a little world where all the preceding args are non-null then do some computation on it!

Reply:
hmm i dont know that i love this.. kinda unreadable and you lose the type checking on the arguments you pass into your fn.. and theres the implicit == anyway in the helper function

Reply:
You don't lose type checking on the arguments. unblock is the type of unblockBlockHeight minus null and undefined, same for actual/height. And we use == everywhere for null checks